src/layers/core/icon-layer/icon-layer.js

```diff
@@ -107,7 +107,14 @@ export default class IconLayer extends Layer {
   updateState({oldProps, props, changeFlags}) {
     super.updateState({props, oldProps, changeFlags});
 
-    const {iconAtlas} = props;
+    const {iconAtlas, iconMapping} = props;
+
+    if (oldProps.iconMapping !== iconMapping) {
```

Comment:
Though in practice, iconMapping and iconAtlas almost always change at the same time. but logically, we might want to do:  if (oldProps.iconMapping !== iconMapping || oldProps.iconAtlas !== iconAtlas) { ...